lib/portage/sync/modules/git/git.py

```diff
@@ -228,6 +228,24 @@ def update(self):
         if not self.verify_head(revision="refs/remotes/%s" % remote_branch):
             return (1, False)
 
+        # Clean up the repo before trying to sync to upstream's
```

Comment:
What is the motivation for doing so? It appears this could cause data loss for ebuild developers with unadded files in the repo.